File: TeamOne/python/rasa/rasa.py
import sys

def count_files():
    filenames = sys.argv[1:]
    complexity = 0
    lines = 0
    for filename in filenames:
        file_complexity, file_lines = count_rasa(filename)
        complexity += file_complexity
        lines += file_lines

    print "total: lines %s, RaSa %s" % (lines, complexity)

def count_rasa(filename):
    multiplier = 1
    complexity = 0
    lines = 0
    with open(filename) as fh:
        for line in fh:
            for char in line:
                if char == "{":
                    multiplier += 1
                elif char == "}":
                    multiplier -= 1
                elif char == ";":
                    complexity += multiplier
            lines += 1
    print "%s: lines %s, RaSa %s" % (filename, lines, complexity)
    return complexity, lines

if __name__ == "__main__":
    count_files()

File: TeamOne/Java/src/RaSa.java
import java.io.*;

/**
 * Created by msavolainen on 12/06/15.
 */
public class RaSa {

	public static void main (String[] args) {
		int totalLines = 0;
		int totalRasa = 0;

		for (String fileName : args) {
			int lines = 0;
			int rasa = 0;
			int multiplier = 1;

			// Open the file
			FileInputStream fstream = null;
			try {
				fstream = new FileInputStream(fileName);
				BufferedReader br = new BufferedReader(new InputStreamReader(fstream));

				String line;

				//Read File Line By Line
				while ((line = br.readLine()) != null)   {
					// calculate line
					for(int i = 0; i < line.length(); i++) {
						switch (line.charAt(i)) {
							case '{':
								multiplier += 1;
								break;
							case '}':
								multiplier -= 1;
								break;
							case ';':
								rasa += multiplier;
								break;
						}
					}

					// increment lines
					lines += 1;
				}

				//Close the input stream
				br.close();
			} catch (FileNotFoundException fnfe) {
				fnfe.printStackTrace();
			} catch (IOException ioe) {
				ioe.printStackTrace();
			}

			System.out.println(fileName + ": lines " + lines + ", RaSa: " + rasa);
			totalLines += lines;
			totalRasa += rasa;
		}

		System.out.println("Total: lines " + totalLines + ", RaSa: " + totalRasa);
	}
}
